src/librustc/middle/check_match.rs

```diff
@@ -680,8 +695,8 @@ fn is_useful(cx: &MatchCheckCtxt,
 
     let constructors = pat_constructors(cx, v[0], left_ty, max_slice_length);
     if constructors.is_empty() {
-        match missing_constructor(cx, matrix, left_ty, max_slice_length) {
-            None => {
+        match &missing_constructors(cx, matrix, left_ty, max_slice_length)[..] {
```

Comment:
Nit: I think I'd prefer:

```rust
let constructors = missing_costructors(...);
if constructors.is_empty() {
    ...
} else {
    ...
}
```

Slice patterns are kind of unreliable and, besides, it has less rightward drift.